gap/TGQuotientSequences.gi

```diff
@@ -426,4 +426,655 @@ function(sequence)
+        	return fail;
+    	fi;
+
+	if not nonMirrorSymmetric and not quotient = 0 then
+		if mspLst[quotient] = 0 then
+			Print("This quotinet does not have mirror symmetries.\n");
+			return fail;
+		fi;
+	fi;	
+
+	# ----------------------------
+	# Restructure adjacency matrix:
+	# ----------------------------
+	
+	if not sparse then
+		adjLst := List(adjMat, row -> SSortedList(Positions(row, 1)));
+	else
+		adjLst := List([1..N], item -> SSortedList([])); 
+		for entry in adjMat do
+			Add(adjLst[entry[1,1]], entry[1,2]);
+		od;
+	fi; 
+
+	# -----------------------------
+	# Apply option mirrorSymmetric:
+	# -----------------------------
+
+	# TODO: optimize
+	if not nonMirrorSymmetric then
+		for i in [1..N] do
+			for j in [1..N] do
+				if mspLst[i] = 0 then
+					adjLst[i] := SSortedList([]);
+					break;
+				elif mspLst[j] = 0 then
+					RemoveSet(adjLst[i], j);
+				fi;
+			od;
+		od;
+	fi;		
+
+	# -------------------
+	# Depth First Search:
+	# -------------------
+
+    	idxLst := List([1..N], item -> [item]);
+    	visited := List([1..N], item -> false);
+
+    	# Call DFS for every unvisited vertex 
+    	for i in [1..N] do
+            	if not visited[i] then 
+            		dfs@(i, adjLst, idxLst, visited);
+		fi;
+	od;
+	
+	if quotient = 0 then
+		return lstTGQuotients{idxLst[PositionMaximum(List(idxLst, item -> Length(item)))]};
+	else 
+		return lstTGQuotients{idxLst[quotient]};
+	fi;
+end );
+
+
+InstallMethod( Export, [  IsTGQuotientSequencesAdjacencyMatrixObj, IsOutputTextStream ],
+function(tgQSAdjMat, output)
+
+    SetPrintFormattingStatus(output, false);
+    
+	# version
+	AppendTo(output, "HyperCells HCQS version 1.0");
+	AppendTo(output, "\n");
+
+    # write: triangle group, genus bound, list of quotients
+	AppendTo(output, Signature(tgQSAdjMat));
+	AppendTo(output, "\n");
+	AppendTo(output, GetBoundByGenus(tgQSAdjMat));
+	AppendTo(output, "\n");
+	AppendTo(output, GetListTGQuotients(tgQSAdjMat));
+	AppendTo(output, "\n");
+	AppendTo(output, GetMirrorSymmetries(tgQSAdjMat));
+	AppendTo(output, "\n");
+	AppendTo(output, IsSparse(tgQSAdjMat));
+	AppendTo(output, "\n");
+
+    # write: adjacency matrix
+	AppendTo(output, GetAdjacencyMatrix(tgQSAdjMat));
+end );
+
+InstallMethod( Export, [ IsTGQuotientSequencesAdjacencyMatrixObj, IsString ],
+function(tgQSAdjMat, path)
+    local output;
+
+    # open file
+    output := OutputTextFile(path, false);
+
+    Export(tgQSAdjMat, output);
+
+    # close file
+    CloseStream(output);
+end );
+
+
+InstallMethod( ExportString, [ IsTGQuotientSequencesAdjacencyMatrixObj ],
+function(tgQSAdjMat)
+	local str, output;
+
+	# open string stream
+	str := "";
+	output := OutputTextString(str, false);
+	
+	# export to stream
+	Export(tgQSAdjMat, output);
+
+	# close
+	CloseStream(output);
+
+	# return
+	return str;
+end );
+
+
+
+InstallGlobalFunction( ImportTGQuotientSequencesAdjacencyMatrix,
+function(input, args...)
```

Comment:
There do not seem to be any optional arguments in use (here and in the subsequent import functions). If that is the case, remove "args...".